### PR TITLE
Revised introductory RDoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Gentler, friendlier introductory material, for new users.

### CODE REVIEW

File: lib/csv.rb [hunk 1]
```diff
@@ -103,85 +103,164 @@
...
 
-# This class provides a complete interface to CSV files and data. It offers
-# tools to enable you to read and write to and from Strings or IO objects, as
-# needed.
+# == \CSV
+# \CSV (comma-separated variables) data is a text representation of a table:
+# - A _row_ _separator_ delimits table rows.
+#   A common row separator is the newline character <tt>"\n"</tt>.
+# - A _column_ _separator_ delimits fields in a row.
+#   A common column separator is the comma character <tt>","</tt>.
```
Comment (BurdetteLamar) [inline lib/csv.rb:111]: We really need to say what a CSV is.

File: lib/csv.rb [hunk 2]
```diff
@@ -103,85 +103,164 @@
...
+# - Generating \CSV data to a \String object.
+#
+# To make \CSV available:
 #   require 'csv'
 #
-# The most generic interface of the library is:
+# All examples here assume that this has been done.
 #
-#    csv = CSV.new(io, **options)
+# == Keeping It Simple
```
Comment (BurdetteLamar) [inline lib/csv.rb:131]: Brief summary of simple uses.

File: lib/csv.rb [hunk 3]
```diff
@@ -103,85 +103,164 @@
...
+#   CSV.filter(in_string, out_string) do |row|
+#     row[0] = row[0].upcase
+#     row[1] *= 4
+#   end
+#   out_string # => "FOO,0000\nBAR,1111\nBAZ,2222\n"
 #
-#   # Core extensions for converting one line
-#   csv_string = ["CSV", "data"].to_csv   # to CSV
-#   csv_array  = "CSV,String".parse_csv   # from CSV
+# == \CSV Objects
```
Comment (BurdetteLamar) [inline lib/csv.rb:256]: Much more needed her, but better not to make this PR too huge.

**Comment by kou** [comment]
Thanks.

**Pull request merged**